R/bed_absdist.r

```diff
@@ -41,32 +45,41 @@
 
 bed_absdist <- function(x, y, genome) {
   
-  x <- group_by(x, chrom, add = TRUE)
-  y <- group_by(y, chrom, add = TRUE)
-  
-  # make sure that y tbl has same grouping as x tbl
-  y <- set_groups(y, x)
+  # find minimum shared groups
+  xy_groups <- shared_groups(y, x)
   
+  x <- group_by_(x, .dots = c("chrom", xy_groups))
+  y <- group_by_(y, .dots = c("chrom", xy_groups))
+
   res <- absdist_impl(x, y)
   
   # calculate reference sizes
-  y_chroms <- unique(y$chrom)
-  y_groups <- purrr::map_chr(groups(y), as.character)
-  
-  genome <- filter(genome, genome$chrom %in% y_chroms)
+  if (!is.null(xy_groups)){
```

Comment:
If there are no shared groups the return value an empty list. Perhaps you should set `groups_xy <- NULL` in `shared_groups` if there are no shared groups. That makes more sense than testing for an empty list. I would also add a test or two that confirms expected behaviour.

``` r
library(dplyr)

x <- tibble::tribble(
  ~chrom, ~start, ~end,
  "chr1", 1     , 100,
  "chr1", 200,    500
)
y <- tibble::tribble(
  ~chrom, ~start, ~end,
  "chr1", 1     , 100,
  "chr1", 200,    500
) %>% group_by(chrom)

intersect(groups(x), groups(y))
#> list()
```